libraries/resource_provider.rb

```diff
@@ -281,6 +281,14 @@ def load_resource(resource_class_type, resource_id, ret_attribute = nil)
         base_module: @sdk_base_module
       )
     end
+
+    # Validates the presence of resource properties
+    # @param [Symbol] property An property name to be validating presence
+    # @param [Symbol] ... More property names
+    # @raise [RuntimeError] if some property is not set
+    def validates_presence_of!(*properties)
```

Comment:
why `!`? 
Conventionally `!` in a method means it will alter the object for which it was called, like `retrieve!` which overwrites the current instance of the class from which it was called.
I suggest dropping this `!` and naming it: `validate_presence_of`. I don't see why we'd use `validates` for a method name as well.

Reply:
Even better, how about `validate_properties` or `validate_required_properties`? I feel that validate presence of something is too generic.

Reply:
The intention of the `!` is because the behavior of this method is throws an exception when some property is not present. But, no problem.

Reply:
> Even better, how about validate_properties or validate_required_properties? I feel that validate presence of something is too generic.

No problem for me. =)